connectors/sources/network_drive.py

```diff
@@ -387,13 +402,18 @@ async def find_matching_paths(self, advanced_rules):
 
     async def ping(self):
         """Verify the connection with Network Drive"""
+
+        if self.smb_connection.session is not None:
+            return
         loop = asyncio.get_running_loop()
-        await loop.run_in_executor(executor=None, func=self.create_connection)
+        await loop.run_in_executor(
+            executor=None, func=self.smb_connection.create_connection
+        )
```

Comment:
Ok I don't think it's working well - we still rely on the fact that `ping` is called before we can call `get_doc`. Can you make it work such that there is no sequential dependency like this?

E.g. making a method `async def get_connection(self)` that creates a connection if it does not exist, otherwise returns existing connection. This way we won't rely on the sequence of method calls to make connector work.

Reply:
I have also removed the singleton approach so now there is no dependency between ping and get_docs. the create_connection method itself handles the case of:  `create a connection if it does not exist, otherwise returns existing connection`,  and create_connection is called at the start of get_docs too. So both ping and get_docs works independently.